snekbox/nsjail.py

```diff
@@ -267,21 +267,32 @@ def python3(
 
             # Parse attachments with time limit
             try:
-                attachments = timed(
-                    MemFS.files_list,
-                    (fs, self.files_limit, self.files_pattern),
-                    {
-                        "preload_dict": True,
-                        "exclude_files": files_written,
-                    },
-                    timeout=self.files_timeout,
-                )
+                with time_limit(self.files_timeout):
+                    attachments = fs.files_list(
+                        limit=self.files_limit,
+                        pattern=self.files_pattern,
+                        preload_dict=True,
+                        exclude_files=files_written,
+                        timeout=self.files_timeout,
+                    )
                 log.info(f"Found {len(attachments)} files.")
+            except RecursionError:
+                log.info("Recursion error while parsing attachments")
+                return EvalResult(
+                    args,
+                    None,
+                    "FileParsingError: Exceeded directory depth limit while parsing attachments",
+                )
             except TimeoutError as e:
                 log.info(f"Exceeded time limit while parsing attachments: {e}")
                 return EvalResult(
                     args, None, "TimeoutError: Exceeded time limit while parsing attachments"
                 )
+            except Exception as e:
+                log.error(f"Unexpected {type(e).__name__} while parse attachments: {e}")
```

Comment:
Use `log.exception` so we get the traceback in logs, which I feel is important for a catch-all.

Reply:
Changed to `log.exception` in [6b0dba5](https://github.com/python-discord/snekbox/pull/173/commits/6b0dba59e1d7a168e97531e5bd16ca04e5ed615a)